src/javaharness/java/arcs/api/Type.java

```diff
@@ -0,0 +1,17 @@
+package arcs.api;
+
+public class Type {
+    public final String tag;
+    // TODO: add more fields.
+
+    public Type(String tag) {
+        this.tag = tag;
+    }
+
+    public boolean isCollection() {
+        return this.tag == "Collection";
```

Comment:
In java. you need to use .equals(), so this.tag.equals("Collection")

But since these are enumerations essentially, should probably declare this as an enum

enum Type  {
   SINGLETON, COLLECTION;
   public boolean isCollection() { return this == COLLECTION; }
   public static Type fromJson(PortableJson json) {
        return "Collection".equalsIgnoreCase(json.getString("tag")) ? COLLECTION : SINGLETON;
   }

Reply:
thanks, i added a Tag enum instead (in the follow up PRs I'll add EntityType and CollectionType classes extending Type, similar to how type.ts does).